File: tests/loop/shift/loop_shift_doc.ml
open Optitrust
open Prelude

let _ = Flags.check_validity := true

let _ = Run.script_cpp (fun _ ->
  !! Loop.shift ~index:"i2" StartAtZero [cFor "i"];
     Loop.shift (ShiftBy (trm_find_var "shift" [])) [cFor "k"];
)

File: tests/loop/shift/loop_shift_doc.cpp
#include <optitrust.h>

void f() {
  int x = 0;
  for (int i = 2; i < 12; i++){
    x += i;
  }
  const int shift = 2;
  for (int k = 0; k < 10; k++){
    x += k;
  }
}
